src/matter/cluster/server/GeneralCommissioningServer.ts

```diff
@@ -12,9 +16,18 @@ export const GeneralCommissioningClusterHandler: ClusterServerHandlers<typeof Ge
         return SuccessResponse;
     },
 
-    updateRegulatoryConfig: async ({request: {breadcrumbStep, config}, attributes: {breadcrumb, regulatoryConfig}}) => {
+    setRegulatoryConfig: async ({request: {breadcrumbStep, newRegulatoryConfig, countryCode}, attributes: {breadcrumb, regulatoryConfig, locationCapability}}) => {
+        const locationCapabilityValue = locationCapability.get();
+        if (locationCapabilityValue !== RegulatoryLocationType.IndoorOutdoor && newRegulatoryConfig !== locationCapabilityValue) {
+            return {errorCode: CommissioningError.ValueOutsideRange, debugText: "Invalid regulatory location"};
+        }
+        if (locationCapabilityValue === RegulatoryLocationType.IndoorOutdoor) {
+            regulatoryConfig.set(newRegulatoryConfig);
+        }
+
+        // TODO countryCode should be set for the BasicInformationCluster.location!
```

Comment:
How can I access the other Clusters ? :-) (Just started to look into an easy use case to see if I understood it)

Reply:
You can't directly... (and this is by design to enforce good separation and the ability to unit test)
Adhoc methods can be added on the server that can be assessed through the session.

Or you can force the creation of the basic cluster first then pass the attribute you need when you construct this cluster:

type GeneralCommissioningClusterConfig = {
    deviceLocation: AttributeServer<string>,
};

export const GeneralCommissioningClusterHandler: (conf:GeneralCommissioningClusterConfig) => ClusterServerHandlers<typeof GeneralCommissioningCluster> = conf => {

Now you can access any require cluster configuration from inside the handlers while still keeping a good separation between clusters.

Reply:
In fact I see several cases where a cluster needs to access other clusters. Like several clusters need to Access the Scene cluster (and then scene cluster needs access again to the others - at least I would do a "please store the device state in scene or restore is later").

The "session" feels the wrong place for it. So yes the idea with a "Conf" will work for some cases but might not for others (Scene) - unless we put this into "session" but session for me is even more global